eng/regenerate-third-party-notices.proj

```diff
@@ -0,0 +1,51 @@
+<Project Sdk="Microsoft.Build.NoTargets">
+
+  <UsingTask TaskName="RegenerateThirdPartyNotices" AssemblyFile="$(InstallerTasksAssemblyPath)" />
+
+  <Target Name="RegenerateThirdPartyNotices" AfterTargets="Build">
+    <PropertyGroup>
+      <TpnFile>$(InstallerProjectRoot)pkg\THIRD-PARTY-NOTICES.TXT</TpnFile>
+    </PropertyGroup>
+
+    <!--
+      Repo configuration. Upstreams, but also more: the TPN in dotnet/runtime serves many repos outside
+      its graph, because dotnet/runtime produces the installer that ends up placing the single TPN file
+      in the dotnet home directory.
+    -->
+    <ItemGroup>
+      <TpnRepo Include="dotnet/runtime" />
+      <TpnRepo Include="dotnet/aspnetcore" />
+      <TpnRepo Include="dotnet/installer" />
+      <TpnRepo Include="dotnet/roslyn-analyzers" />
+      <TpnRepo Include="dotnet/templating" />
+      <TpnRepo Include="dotnet/winforms" />
+      <TpnRepo Include="dotnet/wpf" />
+
+      <!--
+        Additional repos that should be included but don't have any third-party-notices files:
```

Comment:
Do we have an issue which tracks adding these repos?

dotnet/emsdk would be another one.

Reply:
Thanks - I've added this repo and created the tracking issue: https://github.com/dotnet/runtime/issues/61466